File: cpp/base/src/vsdk/toolkit/io/wrapper/Vector3DPrinter.cpp
#include "vsdk/toolkit/io/wrapper/Vector3DPrinter.h"

void
Vector3DPrinter::print(const Vector3D &vector, java::PrintStream *stream) {
    if ( stream == nullptr ) {
        return;
    }
    stream->printf("%g %g %g", vector.x, vector.y, vector.z);
}

File: java/base/src/main/java/vsdk/toolkit/io/wrapper/Vector3DPrinter.java
package vsdk.toolkit.io.wrapper;

import java.io.PrintStream;
import vsdk.toolkit.common.linealAlgebra.Vector3D;

public class Vector3DPrinter {
    public static void print(Vector3D vector, PrintStream stream) {
        if (stream == null) {
            return;
        }
        stream.printf("%g %g %g", vector.x, vector.y, vector.z);
    }
}
